tests/client/api_v1/test_get_info.c

```diff
@@ -1,34 +1,66 @@
+      .port = 14265  // use default port number
   };
-  res_node_info_t info;
+  res_node_info_t* info = res_node_info_new();
+  TEST_ASSERT_NOT_NULL(info);
+
+  // test null cases
+  TEST_ASSERT_EQUAL_INT(-1, get_node_info(NULL, NULL));
+  TEST_ASSERT_EQUAL_INT(-1, get_node_info(&ctx, NULL));
+  TEST_ASSERT_EQUAL_INT(-1, get_node_info(NULL, info));
 
-  int ret = get_node_info(&ctx, &info);
+  int ret = get_node_info(&ctx, info);
   TEST_ASSERT_EQUAL_INT(0, ret);
-  TEST_ASSERT_EQUAL_STRING("HORNET", info.name);
-  TEST_ASSERT_EQUAL_STRING("0.5.3", info.version);
+  TEST_ASSERT_EQUAL_STRING("HORNET", info->u.output_node_info->name);
+  // TEST_ASSERT_EQUAL_STRING("0.6.0-alpha", info->u.output_node_info->version);
+  // TEST_ASSERT_EQUAL_STRING("alphanet1", info->u.output_node_info->network_id);
+  // TEST_ASSERT_EQUAL_UINT64(87389, info->u.output_node_info->latest_milestone_index);
+  // TEST_ASSERT_EQUAL_UINT64(87389, info->u.output_node_info->solid_milestone_index);
+  // TEST_ASSERT_EQUAL_UINT64(0, info->u.output_node_info->pruning_milestone_index);
+  // char **p;
+  // p = NULL;
+  // p = (char**)utarray_next(info->u.output_node_info->features, p);
+  // TEST_ASSERT_EQUAL_STRING("PoW", *p);
+
+  res_node_info_free(info);
 }
 
 void test_deser_node_info() {
   char const* const json_info =
-      "{\"data\":{\"name\":\"HORNET\",\"version\":\"0.5.3\",\"isHealthy\":true,\"networkId\":234,\"latestMilestoneId\":"
-      "\"1a4a9199997db6ec0d6c798040e057df2b505616e5e887257b0600eee49f6345\",\"latestMilestoneIndex\":82847,"
-      "\"solidMilestoneId\":\"1a4a9199997db6ec0d6c798040e057df2b505616e5e887257b0600eee49f6345\","
-      "\"solidMilestoneIndex\":82847,\"pruningIndex\":82325,\"features\":[]}}}}";
+      "{\"data\":{\"name\":\"HORNET\",\"version\":\"0.6.0-alpha\",\"isHealthy\":true,\"networkId\":\"alphanet1\","
+      "\"latestMilestoneIndex\":82847,"
+      "\"solidMilestoneIndex\":82847,\"pruningIndex\":82325,\"features\":[\"feature1\", \"feature2\"]}}}}";
+
+  res_node_info_t* info = res_node_info_new();
+  TEST_ASSERT_NOT_NULL(info);
+
+  int ret = deser_node_info(json_info, info);
 
-  res_node_info_t info = {};
-  int ret = deser_node_info(json_info, &info);
   TEST_ASSERT_EQUAL_INT(0, ret);
-  TEST_ASSERT_EQUAL_STRING("HORNET", info.name);
-  TEST_ASSERT_EQUAL_STRING("0.5.3", info.version);
-  // TODO
+  TEST_ASSERT_EQUAL_STRING("HORNET", info->u.output_node_info->name);
+  TEST_ASSERT_EQUAL_STRING("0.6.0-alpha", info->u.output_node_info->version);
+  TEST_ASSERT_TRUE(info->u.output_node_info->is_healthy);
+  TEST_ASSERT_EQUAL_STRING("alphanet1", info->u.output_node_info->network_id);
+  TEST_ASSERT_EQUAL_UINT64(82847, info->u.output_node_info->latest_milestone_index);
+  TEST_ASSERT_EQUAL_UINT64(82847, info->u.output_node_info->solid_milestone_index);
+  TEST_ASSERT_EQUAL_UINT64(82325, info->u.output_node_info->pruning_milestone_index);
+  char** p;
+  p = NULL;
+  p = (char**)utarray_next(info->u.output_node_info->features, p);
```

Comment:
it's not easy to understand for API users, please provide a method for accessing `features`.

Reply:
done.
I think there are many ways to implement this. Not sure if I did the best way.